### PR TITLE
all: refactor .pb.go generation

### PR DESCRIPTION
**Pull request by neild**
Replace various //go:generate lines and regenerate.sh scripts with a
single, top-level regenerate.sh that regenerates all .pb.go files.

Placing generation in a single script ensures that all files are
generated with similar parameters. The new regenerate.sh uses the
protoc-gen-go version defined in test/tools/go.mod and automatically
handles new .proto files as they are added.

Do some minor refactoring on .proto files: Every file now has a
go_package option (which will be required by a future version of the
code generator), and file imports are all relative to the repository
root.

### CODE REVIEW

File: regenerate.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+# Copyright 2018 gRPC authors.
```
Comment (dfawley) [inline regenerate.sh:2]: 2020?

File: regenerate.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+function finish {
+  rm -rf "$WORKDIR"
+}
+trap finish EXIT
+
+export GOBIN=${WORKDIR}/bin
+export PATH=${GOBIN}:${PATH}
+mkdir -p ${GOBIN}
+
+echo "go install github.com/golang/protobuf/protoc-gen-go"
```
Comment (dfawley) [inline regenerate.sh:29]: IIUC we can now delete

https://github.com/grpc/grpc-go/blob/master/vet.sh#L43

Or, we can make this script assume that has been done, and not override GOBIN.

File: regenerate.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  ${WORKDIR}/grpc-proto/grpc/gcp/altscontext.proto
+  ${WORKDIR}/grpc-proto/grpc/gcp/handshaker.proto
+  ${WORKDIR}/grpc-proto/grpc/gcp/transport_security_common.proto
+  ${WORKDIR}/grpc-proto/grpc/health/v1/health.proto
+  ${WORKDIR}/grpc-proto/grpc/lb/v1/load_balancer.proto
+  ${WORKDIR}/grpc-proto/grpc/lookup/v1/rls.proto
+  ${WORKDIR}/grpc-proto/grpc/service_config/service_config.proto
+)
+SOURCES+=($(
+  git ls-files --exclude-standard --cached --others "*.proto"
```
Comment (dfawley) [inline regenerate.sh:54]: Can this be merged on line 52?

```
...
  ${WORKDIR}/grpc-proto/grpc/service_config/service_config.proto
  $(git ls-files --exclude-standard --cached --others "*.proto")
)
```

File: regenerate.sh (hunk 1 above)
Comment (neild) [reply regenerate.sh]: Done

File: regenerate.sh (hunk 2 above)
Comment (neild) [reply regenerate.sh:29]: Removed protoc-gen-go from vet.sh.

File: regenerate.sh (hunk 3 above)
Comment (neild) [reply regenerate.sh]: done

**Comment by neild** [comment]
Resolved the merge conflict.

Ran regenerate.sh, which also picked up a minor change to `balancer/rls/internal/proto/grpc_lookup_v1/rls.pb.go`.

**Comment by dfawley** [comment]
> Ran regenerate.sh, which also picked up a minor change to
> balancer/rls/internal/proto/grpc_lookup_v1/rls.pb.go.

Unfortunately, this now triggers a deprecated check, because one of the fields was marked as deprecated.  Can you take a look @easwars?

**Comment by easwars** [comment]
> > Ran regenerate.sh, which also picked up a minor change to
> > balancer/rls/internal/proto/grpc_lookup_v1/rls.pb.go.
> 
> Unfortunately, this now triggers a deprecated check, because one of the fields was marked as deprecated. Can you take a look @easwars?

Sent https://github.com/grpc/grpc-go/pull/3562.

**Comment by neild** [comment]
Rebased to after #3562 .

**Pull request merged**